Tools/cases_generator/tier1_generator.py

```diff
@@ -0,0 +1,417 @@
+"""Generate the main interpreter switch.
+Reads the instruction definitions from bytecodes.c.
+Writes the cases to generated_cases.c.h, which is #included in ceval.c.
+"""
+
+import argparse
+import os.path
+import sys
+
+from analyzer import (
+    Analysis,
+    Instruction,
+    Uop,
+    Part,
+    analyze_files,
+    Skip,
+    StackItem,
+    analysis_error,
+)
+from cwriter import CWriter
+from typing import TextIO, Iterator
+from lexer import Token
+from stack import StackOffset
+
+
+HERE = os.path.dirname(__file__)
+ROOT = os.path.join(HERE, "../..")
+THIS = os.path.relpath(__file__, ROOT).replace(os.path.sep, "/")
+
+DEFAULT_INPUT = os.path.relpath(os.path.join(ROOT, "Python/bytecodes.c"))
+DEFAULT_OUTPUT = os.path.relpath(os.path.join(ROOT, "Python/generated_cases.c.h"))
+
+
+def write_header(filename: str, outfile: TextIO) -> None:
+    outfile.write(
+        f"""// This file is generated by {THIS}
+// from:
+//   {filename}
+// Do not edit!
+
+#ifdef TIER_TWO
+    #error "This file is for Tier 1 only"
+#endif
+#define TIER_ONE 1
+"""
+    )
+
+
+FOOTER = "#undef TIER_ONE\n"
+
+
+class SizeMismatch(Exception):
+    pass
+
+
+class Stack:
+    def __init__(self) -> None:
+        self.top_offset: StackOffset = StackOffset()
+        self.base_offset: StackOffset = StackOffset()
+        self.peek_offset: StackOffset = StackOffset()
```

Comment:
These annotations feel somewhat redundant, and mypy still passes if you get rid of them:

```suggestion
        self.top_offset = StackOffset()
        self.base_offset = StackOffset()
        self.peek_offset = StackOffset()
```